### DISCUSSION TITLE
Has anyone able to solve this issue?   One or more invalid file names were detected.  File names must be lowercase, start and end with a letter character, and contain only alphanumeric characters or underscores:

### DISCUSSION THREAD

**Discussion by lci-ash** [dotnet/maui#7166] [General]
AS the name suggest,  this happens only on Mac.
Error details as follows:

/usr/local/share/dotnet/packs/Microsoft.Maui.Resizetizer.Sdk/6.0.200-preview.14.5099/targets/Microsoft.Maui.Resizetizer.targets(9,9): Error: 
            One or more invalid file names were detected.  File names must be lowercase, start and end with a letter character, and contain only alphanumeric characters or underscores: 

        

	 (Name.of.the.Project)

This is how the error appears. There is no indication on what is causing this issue. Font is not an issue here. 
Here is how the project file looks like:
<Project Sdk="Microsoft.NET.Sdk">

	<PropertyGroup>
		<TargetFrameworks>net6.0-android;net6.0-ios</TargetFrameworks>		
		<OutputType>Exe</OutputType>
		<RootNamespace>Hidden</RootNamespace>
		<UseMaui>true</UseMaui>
		<SingleProject>true</SingleProject>
		<ImplicitUsings>enable</ImplicitUsings>

		
		<ApplicationTitle>Hidden</ApplicationTitle>

		
		<ApplicationId>Hidden</ApplicationId>		
		
        
        <ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
        <ApplicationVersion>1</ApplicationVersion>

		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-ios'">14.2</SupportedOSPlatformVersion>		
		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-android'">21.0</SupportedOSPlatformVersion>		
		<Configurations>Debug;Release;QA;Staging;Prod</Configurations>		
		<RunAOTCompilation>False</RunAOTCompilation>
	</PropertyGroup>
    
    <PropertyGroup Condition="$(TargetFramework.Contains('-android')) and '$(Configuration)' != 'Debug'">
        <AndroidKeyStore>True</AndroidKeyStore>
        <AndroidSigningKeyStore>hidden</AndroidSigningKeyStore>
        <AndroidSigningKeyAlias>hidden</AndroidSigningKeyAlias>
        <AndroidSigningKeyPass></AndroidSigningKeyPass>
        <AndroidSigningStorePass></AndroidSigningStorePass>
    </PropertyGroup>
    
    <PropertyGroup Condition="$(TargetFramework.Contains('-ios')) and '$(Configuration)' != 'Debug'">
      <RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
      <CodesignKey>hidden</CodesignKey>
      <CodesignProvision>hidden</CodesignProvision>
    </PropertyGroup>
    
	<ItemGroup>        
		
        <MauiIcon Include="Resources\appicon.png" ForegroundFile="Resources\appicon.png" IsAppIcon="true" Color="#5ec368" />

		
		<MauiSplashScreen Include="Resources\Images\ic_logo_white.png" Color="#e8eaeb" BaseSize="128,128" />
		
		
		<MauiImage Include="Resources\Images\*" />
		<MauiFont Include="Resources\Fonts\*" />

		
		<MauiAsset Include="Resources\Raw\**" LogicalName="%(RecursiveDir)%(Filename)%(Extension)" />
	</ItemGroup>
        
	<ItemGroup>
	  <AndroidAsset Include="hidden" />
	</ItemGroup>

	<ItemGroup>
	  <AndroidResource Remove="Views\**" />
	  <Compile Remove="Views\**" />
	  <EmbeddedResource Remove="Views\**" />
	  <MauiCss Remove="Views\**" />
	  <MauiXaml Remove="Views\**" />
	  <None Remove="Views\**" />
	</ItemGroup>

    <ItemGroup>        
	  <PackageReference Include="Acr.UserDialogs" Version="7.2.0.564" />        
	  <PackageReference Include="Fody" Version="6.6.1">
	    <PrivateAssets>all</PrivateAssets>
	    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
	  </PackageReference>
	  <PackageReference Include="Microsoft.AppCenter" Version="4.5.1" />
	  <PackageReference Include="Microsoft.AppCenter.Crashes" Version="4.5.1" />
	  <PackageReference Include="MonkeyCache.FileStore" Version="2.0.0-beta" />
	  <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
	  <PackageReference Include="PropertyChanged.Fody" Version="4.0.0" />
	</ItemGroup>

    <ItemGroup>
      <Compile Update="Resources\Appresources.Designer.cs">
        <DesignTime>True</DesignTime>
        <AutoGen>True</AutoGen>
        <DependentUpon>Appresources.resx</DependentUpon>
      </Compile>
    </ItemGroup>

    <ItemGroup>
      <EmbeddedResource Update="Resources\Appresources.resx">
        <Generator>ResXFileCodeGenerator</Generator>
        <LastGenOutput>Appresources.Designer.cs</LastGenOutput>
      </EmbeddedResource>
    </ItemGroup>
</Project>

The stupid thing is, it works fine on Windows.

**Comment by HobDev** [comment]
Can you check the name of images in the resources folder. The images should follow the naming as described in the error message. Do necessary changes in the name of the images and try to compile the project again.

**Comment by nextcodelab** [comment]
Sometimes deleting obj and bin will resolve this issue...

**Reply by lci-ash** [reply]
All the above steps tried. The build works on windows, not on Mac. 
![image](https://user-images.githubusercontent.com/102994500/168605156-ed494147-6b49-4358-99c4-efa4678f6981.png)

I have to say I am hugely disappointed. At least show me which files are wrong? Not just a message with an empty space.

**Reply by starkovs** [reply]
Same problem... But I added an image with the name city-23.png and get the same error. After this I delete image city-23.png from the folder /images and now I have the same error but without this image...

Error:
Microsoft.Maui.Resizetizer.Sdk/6.0.300-rc.1.5355/targets/Microsoft.Maui.Resizetizer.targets(469,9): error :

**Comment by Vunda** [comment]
Has the same issue. I inserted two png images with correct names. Now I have this error.
I tried:
- remove images.
- delete bin & obj folder.
- restart VS
- reset all changes on the GIT

Nothing help.

**Comment by NabodevMats** [comment]
Same as @Vunda. Done the same steps. I know the files I copied into the images folder had invalid characters, but I fixed them in finder before using them in the project. Even tried to reinstall visual studio for mac with no success. Still seems to have the invalid name in some "cache" outside of bin and obj.

**Comment by edgedj** [comment]
just hit this myself, caused by a .DS_Store file in my Resources/Fonts folder, deleted that file and it went away.

**Reply by rheza** [reply]
This is the correct way .DS_Store is the problem.

**Reply by alex-ponce** [reply]
Thank you. I keep forgetting about these stupid files!

**Reply by rzmz** [reply]
Ran into the same issue just now. Thank you.

**Reply by fonzane** [reply]
Wow, thank you! I had to open the terminal and use `ls -la` in order to see those evil files. Now, finally, my project works again! :)

**Reply by Nogthings** [reply]
My Resources/Fonts dont have the .DS_Store file and i have the same problem:(

**Reply by Simplicity49** [reply]
I had to recreate a new project the last time I ran into this error because I have not gone too far with my code.
Today, I ran into the same and insisted to find solution online.

Thanks for saving me this time

**Reply by MichelAguilera** [reply]
I migrated my whole project to VSCode with all the MAUI extensions and the problem hasn't come up at all. It's not a perfect solution but it's allowed me to continue with my work.

**Reply by 11622191** [reply]
@Nogthings Try ls -la in your image folder and not in fonts... with terminal... as has said @fonzane

**Reply by redwolf2** [reply]
IMO this should be excluded by the maui image generator. How can such a rookie mistake make it to prod? Also: Why the hell does the output give no details about the file with the problem. If it named the file, the error would be an easy find and fix.

**Comment by RyuX51** [comment]
Seriously? For this phantom we can indeed search a long time… Have my like and my gratitude, sir...

**Comment by polygonsheep** [comment]
Had this issue.

Backup your Images folder (Or whatever folder you think the problem resource was in). 
Then delete the Images folder. 
Try and compile.
Recreate the Images folder and copy files back a few at a time.

BTW, I didn't see any hidden files in that Images folder (but doesn't mean they weren't there)

**Reply by RyuX51** [reply]
This BTW is quite confusing. When I look for hidden files in a folder, I usually know if there are any in there. 🤔

**Reply by polygonsheep** [reply]
There were no hidden files shown in Mac finder with the hidden files option.  What I meant was perhaps Mac OS keeps certain files hidden regardless of the show hidden option.  Or perhaps deleting the folder cleared another cache file somewhere.

**Reply by JairoMarques** [reply]
Same happens with me. Only after  backup contents, delete Images folder and restore them, I could compile...btw, I didn't find any hidden file anywhere.

**Reply by LightningBit** [reply]
wow, thank you! after I backup resources and then re-insert it worked.

**Comment by nextcodelab** [comment]
These are the main problems in .Net Maui Tooling... Sometimes I cannot determine the exact code error, because when there is only one error in code... The error will multiply to hundreds times,  and we cannot determine the single error... There is one time that, just deleting Bin and Obj will not compile again and fail to rebuild. So I need to create another project with the same project name, so that Bin and Obj working again... BUGS IS IN  .NET MAUI TOOLING.

**Comment by KSemenenko** [comment]
I have the same issue I can't run it form Rider for Mac, but I can build the app in AzureDevOps.
To solve it, I delete all project, and then revert changes from git. so it this way I remove all . DS_Store and after manual call  dotnet build it finally works

**Comment by JuQiang** [comment]
Same issues with many guys. I copyied an .png file to Images folder, complile and errors occured. I deleted the Images folder and copy back the default dotnet_bot.svg file, clean all and rebuild solution, still have 3 errors.

Something else still in cache...

**Comment by mrschiele** [comment]
I had this issue too, the error message did not include any image names. 

I think any files staring with a . will not show as the error message never shows file extensions and just adds an empty string to the message. When I added image files that actually have a naming issues the empty string can be observed at the start of the list of images due to the first comma.

End of the error message: `…only alphanumeric characters or underscores: , ExampleImage, ExampleImage2`

I changed the MauiImage rule in the csproj to only include any files ending in **.svg** like this:

`<MauiImage Include="Resources\Images\*.svg" />`

This worked for me.

**Comment by JuQiang** [comment]
THE FINAL SOLUTION.

/usr/local/share/dotnet/packs/Microsoft.Maui.Resizetizer.Sdk/6.0.486/targets/Microsoft.Maui.Resizetizer.targets(9,9): Error: One or more invalid file names were detected.  File names must be lowercase, start and end with a letter character, and contain only alphanumeric characters or underscores: 

App, App.xaml, AppShell, AppShell.xaml, D3MAUI, Hero, MainPage, MainPage.xaml, MauiProgram, D3MAUI.csproj.nuget.dgspec, D3MAUI.csproj.nuget.g, D3MAUI.csproj.nuget.g, .NETCoreApp,Version=v6.0.AssemblyAttributes, D3MAUI.AssemblyInfo, D3MAUI.AssemblyInfoInputs, D3MAUI.assets, D3MAUI.csproj.AssemblyReference, D3MAUI.GeneratedMSBuildEditorConfig, D3MAUI.GlobalUsings.g, project.assets, project.nuget, AppDelegate, Info, Program, launchSettings (D3MAUI)

Try below steps(D3MAUI is my project name, you should replace it with your own prj name. And I changed csproj file to keep net6.0-maccatalyst only, which means I dropped iOS/Android/Windows building packages)

Step 1. Just open mauiimage.inputs file from folder "/Users/{your name}/Projects/D3MAUI/D3MAUI/obj/Debug/net6.0-maccatalyst", you'll found something like "File=bin/Debug/net6.0-maccatalyst/maccatalyst-x64/stderr.log;Link=;BaseSize=;Resize=;TintColor=;Color=;IsAppIcon=;ForegroundScale=;ForegroundFile=
", which means the file "stderr.log" is treated as a "maui image"

Step2. So, for example, right click "stderr.log" from Solution explorer, click "build action" menu item , and click "None" instead of "MauiImage".  Click main menu "Build-Clear All", and "rebuild all".

Step 3. Repeat step1 and step2, until file "mauiimage.inputs" keep only your corresponding image files instead of other non image files. 

Step 4. You'll find the last compile error: Error: One or more invalid file names were detected.  File names must be lowercase, start and end with a letter character, and contain only alphanumeric characters or underscores: ,

file name "," means a hidden file ds_store

Step 5. open your terminal and switch to corresponding csproj folder, type below command
    find . -name ".DS_Store"
u'll find some hidden .DS_Store files. 
And type  below comand
    find . -name ".DS_Store" -type f -delete
all hidden ds_store files will be deleted. 

Step 6. Click main menu "Build-clean all" and "Rebuild"

It's OK now.

**Reply by JuQiang** [reply]
Or, select all files except your maui images, right click, select build actions-None instead of Maui Image. Then click main menu build-clear all and rebuild.

**Reply by dylan-sacco** [reply]
Worked Thank you

**Comment by vhugogarcia** [comment]
For me the issue was due to the .DS_Store files macOS creates on certain folders.

I executed this line on my terminal and then I re-build the app and worked very good.

`find . -name ".DS_Store" -type f -delete`

**Note**: You need to navigate to the path where your project is located in the terminal and then execute the line above.

**Reply by ShadowOfPhantom** [reply]
Helped me

**Reply by KieranMaclagan** [reply]
Worked for me too, thanks @vhugogarcia!

**Reply by Kremed** [reply]
Worked for me ,thank you

**Reply by navkar** [reply]
Works on a M1 Mac Pro. Thank u.

**Reply by albrecht-jun-zhang** [reply]
Works on my M1 Macbook Pro as well!

**Reply by moneosi** [reply]
Work fine now, thanks

**Reply by dylan-sacco** [reply]
Worked Thanks

**Comment by rsbepvb** [comment]
Any additional ideas or thoughts or updates on this?  The ideas above did not resolve these errors for us.

**Reply by vishalmgiri** [reply]
For me on Windows also I received same error.
From above suggestions I reviewed Resources\Images folder and noticed one unnecessary apk file was present (i might have copied it accidently). After removing it error gone away.

**Reply by mpa74** [reply]
For me,  i found in mauiimage for maccatalyst link to ANDROID platform file  _CleanIntermediateIfNeeded. So, i delete every platforms exceed maccatalyst, but also same error (it was 3, now - 1)

**Reply by mpa74** [reply]
Finally, i create new solution and copy all the project files - it working normally

**Comment by sej69** [comment]
Had I actually read the error message I would have figured it out quickly...  "start with a lowercase..."  The image file I brough in started with an uppercase.  Obviously when I changed this to lowercase it went away.

**Comment by gordonkenyon** [comment]
Here is the answer : It worked for me 
https://stackoverflow.com/questions/72414708/resizetizer-one-or-more-invalid-file-names-were-detected

**Reply by Nogthings** [reply]
works for me

**Reply by jtorvald** [reply]
Thanks

**Comment by JoacimWall** [comment]
Hi 
I reported this to Microsoft in Feb
https://github.com/dotnet/maui/issues/13452
//Joacim

**Comment by nextcodelab** [comment]
My solution to all of these,,  is rewriting the app from maui to flutter...
Flutter super fast almost 1 second in startup... any designs is posible;
[https://play.google.com/store/apps/details?id=com.nxtccode.biblenotebook](https://play.google.com/store/apps/details?id=com.nxtccode.biblenotebook)

**Reply by hinnerkweiler** [reply]
Maybe a bit less drama-queen but `rm -f .DS_Store` in the corresponding folders is a bit quicker than dumping whole projects.
